Double clicking the title bar when maximized doesn't always restore the window size

### Description

When the application is maximized (via the maximize button, not F11) double clicking on the title bar will restore the window to its previous size, but if you click on the top few pixels of the title bar nothing happens. That area will trigger the menus and min/restore/close buttons, but it doesn't trigger the window restore.

### Version

**GitHub Desktop version:** 0.5.5
**OS version:** Win10 Creators Update (10.0.15063)

### Steps to Reproduce

1. Maximize the window so it's full screen (not the F11 option)
1. Move your mouse pointer to the very top of your screen
1. Double click

**Expected behaviour:** The window should be restored to its previous size just like when you double click in the middle of the title bar.

**Actual behaviour:** Nothing.

**Reproduces how often:** Every time.

#### Additional Information

It looks like `#desktop-app-title-bar .resize-handle.top` should be hidden when the window is maximized. Hiding it with dev tools looks to have solved the problem.

So far I haven't had the best of luck getting my environment setup. Despite having node, npm, and VS2015 installed I still needed to do some extra things. This is a list of updates that should be added to the docs once it's all working:

- When installing python make sure to check the option `Add python.exe to Path`
- Even with VS 2015 already installed I still needed to run `npm config set msvs_version 2015`
- Even with VS 2015 & 2017 installed I still needed to install the Win8.1 SDK (I'm on Win10 Creators Update)
    - Only the SDK option is needed
    - https://developer.microsoft.com/en-us/windows/downloads/windows-8-1-sdk
- Even with VS 2015 installed I needed to update the config and install c++ (it's not installed by default)

At this point I was able to `npm install` without errors, but now `npm run build:dev` is giving an error:
```powershell
C:\dev\GitHub\desktop [master ↑6]> npm run build:dev

> @ build:dev C:\dev\GitHub\desktop
> npm run compile:dev && cross-env NODE_ENV=development node script/build


> @ compile:dev C:\dev\GitHub\desktop
> cross-env NODE_ENV=development parallel-webpack --config app/webpack.development.js

[WEBPACK] Building 4 targets
[WEBPACK] Started building renderer.js
[WEBPACK] Started building main.js
[WEBPACK] Started building shared.js
[WEBPACK] Started building ask-pass.js

[at-loader] Using typescript@2.3.2 from typescript and "tsconfig.json" from C:\dev\GitHub\desktop/tsconfig.json.


[at-loader] Using typescript@2.3.2 from typescript and "tsconfig.json" from C:\dev\GitHub\desktop/tsconfig.json.


[at-loader] Using typescript@2.3.2 from typescript and "tsconfig.json" from C:\dev\GitHub\desktop/tsconfig.json.


[at-loader] Using typescript@2.3.2 from typescript and "tsconfig.json" from C:\dev\GitHub\desktop/tsconfig.json.


[at-loader] Checking started in a separate process...

[at-loader] Checking started in a separate process...

[at-loader] Checking started in a separate process...

[at-loader] Ok, 2.451 sec.
Hash: 4d9c182f5dd963840b1a
Version: webpack 2.5.1
Time: 5409ms
      Asset    Size  Chunks             Chunk Names
keytar.node  110 kB          [emitted]
ask-pass.js   15 kB       0  [emitted]  ask-pass
chunk    {0} ask-pass.js (ask-pass) 4.44 kB [entry] [rendered]
    [0] ./app/src/ask-pass/ask-pass.ts 1.77 kB {0} [built]
    [1] ./app/~/keytar/lib/keytar.js 1.39 kB {0} [built]
    [2] ./app/src/ask-pass/main.ts 367 bytes {0} [built]
    [3] ./app/src/lib/auth.ts 412 bytes {0} [built]
    [4] ./app/src/shared-process/token-store.ts 446 bytes {0} [built]
    [5] ./app/~/keytar/build/Release/keytar.node 57 bytes {0} [built]
[WEBPACK] Finished building ask-pass.js within 5.409 seconds

[at-loader] Ok, 2.559 sec.

[at-loader] Ok, 2.598 sec.
Hash: 28404499ae4e87db045b
Version: webpack 2.5.1
Time: 6698ms
      Asset       Size  Chunks                    Chunk Names
keytar.node     110 kB          [emitted]
  shared.js    1.24 MB       0  [emitted]  [big]  shared
 error.html    1.34 kB          [emitted]
shared.html  182 bytes          [emitted]
chunk    {0} shared.js (shared) 463 kB [entry] [rendered]
    [5] ./app/src/lib/api.ts 19.9 kB {0} [built]
    [6] ./app/src/models/account.ts 1.16 kB {0} [built]
   [12] ./app/src/lib/logging/renderer.ts 1.6 kB {0} [built]
   [13] ./app/src/models/repository.ts 1.53 kB {0} [built]
   [14] ./app/~/dexie/dist/dexie.js 199 kB {0} [built]
   [18] ./app/~/octokat/dist/node/plugins/authorization.js 1.71 kB {0} [built]
   [35] ./app/src/shared-process/accounts-store.ts 3.81 kB {0} [built]
   [36] ./app/src/shared-process/communication.ts 2.77 kB {0} [built]
   [37] ./app/src/shared-process/database.ts 627 bytes {0} [built]
   [38] ./app/src/shared-process/repositories-store.ts 9.72 kB {0} [built]
   [39] ./app/src/shared-process/token-store.ts 446 bytes {0} [built]
   [40] ./app/src/ui/lib/exception-reporting.ts 2.42 kB {0} [built]
   [87] ./app/src/lib/2fa.ts 1.15 kB {0} [built]
   [90] ./app/src/lib/fatal-error.ts 777 bytes {0} [built]
   [98] ./app/src/shared-process/index.ts 5.3 kB {0} [built]
     + 91 hidden modules
Child html-webpack-plugin for "shared.html":
          Asset     Size  Chunks  Chunk Names
    shared.html  1.42 MB       0
    chunk    {0} shared.html 541 kB [entry] [rendered]
        [0] ./~/lodash/lodash.js 540 kB {0} [built]
        [1] ./~/html-webpack-plugin/lib/loader.js!./~/html-webpack-plugin/default_index.ejs 538 bytes {0} [built]
        [2] (webpack)/buildin/module.js 517 bytes {0} [built]
Child html-webpack-plugin for "error.html":
         Asset     Size  Chunks  Chunk Names
    error.html  1.42 MB       0
    chunk    {0} error.html 542 kB [entry] [rendered]
        [0] ./~/lodash/lodash.js 540 kB {0} [built]
        [1] ./~/html-webpack-plugin/lib/loader.js!./app/static/error.html 1.8 kB {0} [built]
        [2] (webpack)/buildin/module.js 517 bytes {0} [built]
[WEBPACK] Finished building shared.js within 6.698 seconds
Hash: d43bcf3e9d773cee9024
Version: webpack 2.5.1
Time: 7271ms
  Asset     Size  Chunks                    Chunk Names
main.js  1.36 MB       0  [emitted]  [big]  main
chunk    {0} main.js (main) 485 kB [entry] [rendered]
    [0] external "path" 42 bytes {0} [not cacheable]
    [3] external "electron" 42 bytes {0} [not cacheable]
   [22] ./app/src/lib/fatal-error.ts 777 bytes {0} [built]
   [23] ./app/src/lib/logging/main.ts 3.75 kB {0} [built]
   [44] ./app/~/electron-debug/index.js 2.19 kB {0} [built]
   [45] ./app/~/electron-devtools-installer/dist/index.js 4.63 kB {0} [built]
   [46] ./app/src/lib/parse-url.ts 1.98 kB {0} [built]
   [47] ./app/src/main-process/app-window.ts 9.89 kB {0} [built]
   [48] ./app/src/main-process/error-page.ts 1.94 kB {0} [built]
   [49] ./app/src/main-process/menu/index.ts 296 bytes {0} [built]
   [50] ./app/src/main-process/squirrel-updater.ts 1.17 kB {0} [built]
   [51] ./app/src/shared-process/shared-process.ts 2.27 kB {0} [built]
   [73] ./app/~/electron-localshortcut/index.js 3.33 kB {0} [built]
   [74] ./app/~/electron-window-state/index.js 4.6 kB {0} [built]
  [124] ./app/src/main-process/main.ts 8.41 kB {0} [built]
     + 119 hidden modules

WARNING in ./app/~/colors/lib/colors.js
127:29-43 Critical dependency: the request of a dependency is an expression
[WEBPACK] Finished building main.js within 7.271 seconds

[at-loader] Checking started in a separate process...

[at-loader] Ok, 3.322 sec.
Hash: b7c57f144baad7c93133
Version: webpack 2.5.1
Time: 14218ms
      Asset       Size  Chunks                    Chunk Names
renderer.js    10.3 MB       0  [emitted]  [big]  renderer
 index.html  234 bytes          [emitted]
chunk    {0} renderer.js (renderer) 3.78 MB [entry] [rendered]
    [0] ./app/~/react/react.js 56 bytes {0} [built]
    [3] external "path" 42 bytes {0} [not cacheable]
  [288] ./app/src/ui/app.tsx 41.4 kB {0} [built]
  [305] (webpack)/buildin/module.js 517 bytes {0} [built]
  [308] external "querystring" 42 bytes {0} [not cacheable]
  [310] ./app/~/webpack-hot-middleware/client.js?path=http://localhost:3000/__webpack_hmr 6.68 kB {0} [built]
  [311] ./app/src/ui/index.tsx 4.23 kB {0} [built]
  [585] ./app/~/strip-ansi/index.js 161 bytes {0} [built]
  [595] ./app/~/webpack-hot-middleware/client-overlay.js 1.82 kB {0} [built]
  [596] ./app/~/webpack-hot-middleware/process-update.js 3.88 kB {0} [built]
  [598] ./app/~/wicg-focus-ring/dist/focus-ring.js 8.21 kB {0} [built]
  [710] ./app/src/ui/install-globals.ts 521 bytes {0} [built]
  [716] ./app/src/ui/lib/exception-reporting.ts 2.42 kB {0} [built]
  [774] ./app/styles/desktop.scss 1.11 kB {0} [built]
  [781] multi webpack-hot-middleware/client?path=http://localhost:3000/__webpack_hmr ./app/src/ui/index 40 bytes {0} [built]
     + 767 hidden modules
Child html-webpack-plugin for "index.html":
         Asset     Size  Chunks  Chunk Names
    index.html  1.44 MB       0
    chunk    {0} index.html 541 kB [entry] [rendered]
        [0] ./~/lodash/lodash.js 540 kB {0} [built]
        [1] ./~/html-webpack-plugin/lib/loader.js!./app/static/index.html 529 bytes {0} [built]
        [2] (webpack)/buildin/module.js 517 bytes {0} [built]
[WEBPACK] Finished building renderer.js within 14.218 seconds
[WEBPACK] Finished build after 15.314 seconds
Building for development…
Removing old distribution…
Copying dependencies…
  Installing 7zip (dependency for electron-devtools-installer)
  Copying git environment…
fs.js:968
  binding.stat(pathModule._makeLong(path), statValues);
          ^

Error: ENOENT: no such file or directory, stat 'C:\dev\GitHub\desktop\app\node_modules\dugite\git'
    at Object.fs.statSync (fs.js:968:11)
    at Object.statSync (C:\dev\GitHub\desktop\node_modules\graceful-fs\polyfills.js:297:22)
    at Object.copySync (C:\dev\GitHub\desktop\node_modules\fs-extra\lib\copy-sync\copy-sync.js:31:86)
    at copyDependencies (C:\dev\GitHub\desktop\script\build:204:6)
    at Object.<anonymous> (C:\dev\GitHub\desktop\script\build:23:1)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:427:7)
    at startup (bootstrap_node.js:151:9)
    at bootstrap_node.js:542:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ build:dev: `npm run compile:dev && cross-env NODE_ENV=development node script/build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ build:dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\brian\AppData\Roaming\npm-cache\_logs\2017-05-20T01_48_10_919Z-debug.log
```

--- a/app/src/ui/app.tsx
+++ b/app/src/ui/app.tsx
@@ -586,6 +586,7 @@ export class App extends React.Component<IAppProps, IAppState> {
   private renderTitlebar() {
 
     const inFullScreen = this.state.windowState === 'full-screen'
+    const isMaximized = this.state.windowState === 'maximized'
 
     const menuBarActive = this.state.currentFoldout &&
       this.state.currentFoldout.type === FoldoutType.AppMenu
@@ -610,12 +611,12 @@ export class App extends React.Component<IAppProps, IAppState> {
     // window controls need to disable dragging so we add a 3px tall element which
     // disables drag while still letting users drag the app by the titlebar below
     // those 3px.
-    const topResizeHandle = __WIN32__
+    const topResizeHandle = __WIN32__ && !isMaximized
       ? <div className='resize-handle top' />
       : null
 
     // And a 3px wide element on the left hand side.
-    const leftResizeHandle = __WIN32__
+    const leftResizeHandle = __WIN32__ && !isMaximized
       ? <div className='resize-handle left' />
       : null
 

